alchemist-incarnation-collektive/src/main/kotlin/it/unibo/collektive/alchemist/device/CollektiveDevice.kt

```diff
@@ -49,43 +49,40 @@ class CollektiveDevice<P>(
             neighboringViaExchange(nodePosition).map { position -> nodePosition.distanceTo(position) }
         }
 
+    @Suppress("UNCHECKED_CAST")
+    override fun <T> sense(name: String): T = node.getConcentration(SimpleMolecule(name)) as T
+
+    override fun <T> senseOrElse(name: String, default: T): T = sense(name) ?: default
```

Comment:
Beware that `null` can be a valid concentration, so it would be better to do the following:

```kotlin
override fun <T> senseOrElse(name: String, default: T): T = if (node.contains(SimpleMolecule(name))) sense(name) else default
```